File: sdk/java/examples/RetryExample.java
import com.ordoengine.sdk.OrdoClient;
import com.ordoengine.sdk.config.RetryConfig;
import com.ordoengine.sdk.model.ExecuteResult;

import java.time.Duration;
import java.util.Map;

public class RetryExample {
    public static void main(String[] args) {
        // Client with retry and multi-tenancy
        OrdoClient client = OrdoClient.builder()
                .httpAddress("http://localhost:8080")
                .tenantId("tenant-abc")
                .retry(RetryConfig.builder()
                        .maxAttempts(3)
                        .initialInterval(Duration.ofMillis(100))
                        .maxInterval(Duration.ofSeconds(5))
                        .jitter(true)
                        .build())
                .build();

        // Requests automatically retry on 5xx / 429 / connection errors
        ExecuteResult result = client.execute("risk_check", Map.of(
                "user_id", "u123",
                "action", "transfer"
        ));
        System.out.printf("Result: %s - %s%n", result.getCode(), result.getMessage());

        client.close();
    }
}

File: sdk/python/examples/retry_example.py
"""Retry and multi-tenancy example."""

from ordo import OrdoClient, RetryConfig

# Client with retry and multi-tenancy
client = OrdoClient(
    http_address="http://localhost:8080",
    tenant_id="tenant-abc",
    retry=RetryConfig(
        max_attempts=3,
        initial_interval=0.1,  # 100ms
        max_interval=5.0,      # 5s
        jitter=True,
    ),
)

# Requests will automatically retry on 5xx / 429 / connection errors
result = client.execute("risk_check", {"user_id": "u123", "action": "transfer"})
print(f"Result: {result.code} - {result.message}")

# Context manager usage (auto-close)
with OrdoClient(
    http_address="http://localhost:8080",
    retry=RetryConfig(max_attempts=5),
) as c:
    health = c.health()
    print(f"Status: {health.status}")
